src/operator/controllers/intents_reconcilers/internet_network_policy/internet_network_policy.go

```diff
@@ -0,0 +1,368 @@
+	"strings"
+)
+
+// The InternetNetworkPolicyReconciler creates network policies that allow egress traffic from pods.
+type InternetNetworkPolicyReconciler struct {
+	client.Client
+	Scheme                      *runtime.Scheme
+	RestrictToNamespaces        []string
+	enableNetworkPolicyCreation bool
+	enforcementDefaultState     bool
+	injectablerecorder.InjectableRecorder
+}
+
+func NewInternetNetworkPolicyReconciler(
+	c client.Client,
+	s *runtime.Scheme,
+	restrictToNamespaces []string,
+	enableNetworkPolicyCreation bool,
+	enforcementDefaultState bool) *InternetNetworkPolicyReconciler {
+	return &InternetNetworkPolicyReconciler{
+		Client:                      c,
+		Scheme:                      s,
+		RestrictToNamespaces:        restrictToNamespaces,
+		enableNetworkPolicyCreation: enableNetworkPolicyCreation,
+		enforcementDefaultState:     enforcementDefaultState,
+	}
+}
+
+func (r *InternetNetworkPolicyReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Result, error) {
+	intents := &otterizev1alpha3.ClientIntents{}
+	err := r.Get(ctx, req.NamespacedName, intents)
+	if k8serrors.IsNotFound(err) {
+		return ctrl.Result{}, nil
+	}
+	if err != nil {
+		return ctrl.Result{}, errors.Wrap(err)
+	}
+	if intents.Spec == nil {
+		return ctrl.Result{}, nil
+	}
+
+	err = r.removeOrphanNetworkPolicies(ctx)
+	if err != nil {
+		r.RecordWarningEventf(intents, consts.ReasonRemovingEgressNetworkPolicyFailed, "failed to remove network policies: %s", err.Error())
+		return ctrl.Result{}, errors.Wrap(err)
+	}
+
+	logrus.Infof("Reconciling internet network policies for service %s in namespace %s",
+		intents.Spec.Service.Name, req.Namespace)
+
+	if !intents.DeletionTimestamp.IsZero() {
+		return r.handleDeletion(ctx, intents)
+	}
+
+	if len(r.RestrictToNamespaces) != 0 && !lo.Contains(r.RestrictToNamespaces, intents.Namespace) {
+		r.RecordWarningEventf(intents, consts.ReasonNamespaceNotAllowed, "ClientIntents are in namespace %s but namespace is not allowed by configuration", intents.Namespace)
+		return ctrl.Result{}, nil
+	}
+
+	if !r.enforcementDefaultState {
+		logrus.Infof("Enforcement is disabled globally skipping internet network policy creation for service %s in namespace %s", intents.Spec.Service.Name, req.Namespace)
+		r.RecordNormalEventf(intents, consts.ReasonEnforcementDefaultOff, "Enforcement is disabled globally, internet network policy creation skipped")
+		return ctrl.Result{}, nil
+	}
+	if !r.enableNetworkPolicyCreation {
+		logrus.Infof("Network policy creation is disabled, skipping internet network policy creation for service %s in namespace %s", intents.Spec.Service.Name, req.Namespace)
+		r.RecordNormalEvent(intents, consts.ReasonEgressNetworkPolicyCreationDisabled, "Network policy creation is disabled, internet network policy creation skipped")
+		return ctrl.Result{}, nil
+	}
+
+	hasAnyInternetIntents := slices.ContainsFunc(intents.GetCallsList(), func(intent otterizev1alpha3.Intent) bool {
```

Comment:
I would move this to the top of the function to avoid unnecessary work. You can also use `GetFilteredCallsList`

Reply:
Fixed, moved it up but only after handling `removeOrphan` since it should be handled in any case